test-integration/helpers/test-element.coffee

```diff
@@ -48,26 +51,37 @@ class TestItemHelper
 
   isPresent: (args...) =>
     locator = @getLocator(args...)
-    @test.driver.isElementPresent(locator).then (isPresent) ->
-      isPresent
+    @test.driver.isElementPresent(locator)
+
+  isDisplayed: (args...) =>
+    @isPresent(args...).then (isPresent) =>
+      if isPresent
+        locator = @getLocator(args...)
+        el = @findElement(args...)
+        el.isDisplayed()
+      else
+        false
 
   # Helper for the common case of `get(...).click()`.
   # Plus, it allows a place to add logging since this is one of the most
   # common places for Selenium to time out (trying to click on an element)
   click: (args...) =>
     locator = @getLocator(args...)
+    # Scroll to the element so it is visible before clicking (this assumes `position: fixed` is overridden for all element)
+    # @isDisplayed(args...).then (isDisplayed) =>
+    #   unless isDisplayed
+    #     el = @findElement(args...)
+    #     @test.utils.windowPosition.scrollTo(el)
     @test.utils.verboseWrap "Clicking #{JSON.stringify(locator)}", =>
-      @get(args...).click()
+      el = @get(args...)
+      @test.utils.windowPosition.scrollTo(el)
+      el.click()
 
-# Using defined properties for access eliminates the possibility
-# of accidental assignment
-Object.defineProperties TestItemHelper.prototype,
-  test:
-    get: -> @_test
-  locator:
-    get: -> @_locator
-  element:
-    get: -> @get()
+  waitClick: (args...) =>
+    locator = @getLocator(args...)
+    @test.utils.verbose "Waiting for #{JSON.stringify(locator)}"
+    @test.utils.wait.for(locator)
+    @click(args...)
 
```

Comment:
I guess I figured most of the lines in tests would be `el.click()`, and `el.waitClick()` would be less common, but maybe still useful? Something like:

```coffee
{someButton, saveButton, anotherButton} = foo.el
calendar.goToFoo()  # which implicitly calls `waitUntilLoaded()` because people forget ; )
someButton.click()
anotherButton.click()  # With these being the common thing that happens
saveButton.waitClick()
```